pkg/scheduler/factory.go

```diff
@@ -149,8 +131,19 @@ func (c *Configurator) create() (*Scheduler, error) {
 
 	// The nominator will be passed all the way to framework instantiation.
 	nominator := internalqueue.NewPodNominator()
-	profiles, err := profile.NewMap(c.profiles, c.buildFramework, c.recorderFactory,
-		frameworkruntime.WithPodNominator(nominator))
+	for _, p := range c.profiles {
+		if c.frameworkCapturer != nil {
```

Comment:
Would it be better to put this logic after this check https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/profile/profile.go#L63 ?

Reply:
That might be ideal, but I prefer we don't pass yet another parameter to `NewMap`. Maybe we could just do this loop after calling `NewMap`.

Reply:
I think it's depending on struct `c` and `c.frameworkCapturer()` is a private function.

Reply:
> Would it be better to put this logic after this check https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/profile/profile.go#L63 ?

I think it's depending on struct `c` and `c.frameworkCapturer()` is a private function.